pkg/controller/vsphere/reconciler.go

```diff
@@ -560,6 +576,18 @@ func clone(s *machineScope) (string, error) {
 		return "", handleVSphereError(multipleFoundMsg, notFoundMsg, defaultError, err)
 	}
 
+	hwVersion, err := getHwVersion(s.Context, vmTemplate)
+	if err != nil {
+		// TODO throw an event also?
+		klog.Warningf("Unable to detect machine template HW version for machine %s: %v", s.machine.GetName(), err)
```

Comment:
If the hw version cant be retrieved, this is likely due to a condition which will result in failures later in this process.  It is likely safe to return an error here.

Reply:
TBH i'm bit concerned due to this version came from from string. Is it possible to encounter smth other than "vmx-X" there?